### PR TITLE
A different approach to library wrapper generation and testing

### PR DESCRIPTION
**Pull request by jpsim**
@norio-nomura I *think* this simplifies the wrapper generation process, but I'll let you decide.

Most of your documentation in the README still applies, but I'm not sure where that would go. Any thoughts?

### CODE REVIEW

**Comment by norio-nomura** [comment]
Wow! Nice! It looks much better! :smile:
I'll review!

File: Source/SourceKittenFramework/Request.swift [hunk 1]
```diff
@@ -340,3 +340,70 @@ extension Request: CustomStringConvertible {
     /// A textual representation of `Request`.
     public var description: String { return String(UTF8String: sourcekitd_request_description_copy(sourcekitObject))! }
 }
+
+private func interfaceForModule(module: String) -> [String: SourceKitRepresentable] {
+    let xctestConfigurationPath = NSProcessInfo.processInfo().environment["XCTestConfigurationFilePath"]!
+    let buildPath = xctestConfigurationPath.substringToIndex(xctestConfigurationPath.rangeOfString("/Intermediates/")!.startIndex)
```
Comment (norio-nomura) [inline Source/SourceKittenFramework/Request.swift]: I got crash at here on testing.
It seems my `xctestConfigrationPath` does not contains `/Intermediates/`.
screenshot 2016-03-17 14 41 12

Comment (norio-nomura) [inline Source/SourceKittenFramework/Request.swift]: I ran test of `sourcekitten` scheme inside `SwiftLint.xcworkspace` on above crash.

Comment (norio-nomura) [inline Source/SourceKittenFramework/Request.swift]: On that condition, `XCTestConfigurationFilePath` is `/Users/norio/Library/Developer/Xcode/DerivedData/SwiftLint-eangswdzvhtbxpebxskdhgwvjger/Build/Products/Debug/SourceKittenFrameworkTests.xctest/Contents/Resources/SourceKittenFrameworkTests-BFFF5841-89B0-4790-A8D7-61359F29A156.xctestconfiguration`

On running test of sourcekitten scheme inside `SourceKitten.xcworkspace`, `XCTestConfigurationFilePath` is `/Users/norio/Library/Developer/Xcode/DerivedData/SourceKitten-drsebuetmrpoaedcrikulbrsfzey/Build/Products/Debug/SourceKittenFrameworkTests.xctest/Contents/Resources/SourceKittenFrameworkTests-E6113DED-C71F-471C-931D-47462DC42DEF.xctestconfiguration`

Comment (jpsim) [inline Source/SourceKittenFramework/Request.swift]: Do the tests pass when they're run from `SourceKitten.xcworkspace` at least? Do you know of any other way to get the derived data directory for SourceKittenFramework? Or another way to get all the necessary compiler arguments to SourceKit for interface generation?

Comment (jpsim) [inline Source/SourceKittenFramework/Request.swift]: Never mind, I think I fixed it by passing the compiler arguments from SourceKittenFramework to the interface calls.

**Comment by norio-nomura** [comment]
Looks good to me, but why does Travis-CI not work?
For testing https://github.com/realm/SwiftLint/pull/555 with using this, I added `0.11.0-test` tag to 1ba772a.

**Comment by jpsim** [comment]
> Looks good to me, but why does Travis-CI not work?

I think Travis just runs PRs targeted at master.

If you're ok with this, feel free to merge into your branch and I'll continue to review tomorrow.

**Comment by norio-nomura** [comment]
I understand. Thanks! :pray:

**Pull request merged**